code/addons/viewport/src/Tool.tsx

```diff
@@ -164,6 +164,11 @@ export const ViewportTool: FC = memo(
 
     useEffect(() => {
       ref.current = styles;
+      const iframe = document.querySelector<HTMLIFrameElement>('iframe[data-is-storybook="true"]');
+      const body = iframe?.contentDocument?.querySelector<HTMLBodyElement>('body');
```

Comment:
We should not be hooking into the iframe like this.
The iframe may not be on the same origin/domain, and so this would be blocked by the browser for security.

Architecturally, it also makes little sense for this Tool component to be in charge of changing the iframe inner CSS.

A better solution would be to add a decorator listening to storybook-`globals` that then sets the appropriate styles on the storybook root element.